services/blockchain-indexer/shared/dataService/business/pos/stakers.js

```diff
@@ -93,36 +118,50 @@ const getStakers = async params => {
 		},
 		['stakerAddress', 'amount'],
 	);
-	if (resultSet.length) stakers.data.stakers = resultSet;
 
-	stakers.data.stakers = await BluebirdPromise.map(
-		stakers.data.stakers,
+	// Populate stakers name and prepare response
+	stakersResponse.data.stakers = await BluebirdPromise.map(
+		stakes,
 		async stake => {
-			const { name = null } = await getIndexedAccountInfo({ address: stake.stakerAddress, limit: 1 }, ['name']);
+			// Try to use cached staker name. Query DB if not found.
+			let stakerName = allowedStakersAddressNameMap[stake.stakerAddress];
+			if (!stakerName) {
+				const accountInfo = await getIndexedAccountInfo({
+					address: stake.stakerAddress,
+					limit: 1,
+				},
+				['name'],
+				);
+				stakerName = accountInfo.name;
+			}
+
 			return {
```

Comment:
I think, this logic (`stakersResponse.data.stakers = await BluebirdPromise.map`) doesn't filter out the stakers based on the `search` criteria. Please verify 🤔

Reply:
Actually we are not doing the filtering here. The filtering is done when we fetch list of stakes from the stakes table. we figure out the valid staker addresses for the specified `search` param and filter the stakes based on the while querying the db.